xds/internal/resolver/serviceconfig_test.go

```diff
@@ -106,6 +107,35 @@ func (s) TestGenerateRequestHash(t *testing.T) {
 				Method:  "/some-method",
 			},
 		},
+		// Tests that bin headers are skipped.
+		{
+			name: "skip-bin",
+			hashPolicies: []*xdsresource.HashPolicy{{
+				HashPolicyType: xdsresource.HashPolicyTypeHeader,
+				HeaderName:     "something-bin",
+			}, {
+				HashPolicyType: xdsresource.HashPolicyTypeChannelID,
+			}},
+			requestHashWant: channelID,
+			rpcInfo: iresolver.RPCInfo{
+				Context: metadata.NewOutgoingContext(context.Background(), metadata.Pairs("something-bin", "xyz")),
+			},
+		},
+		// Tests that extra metadata is used first.
```

Comment:
Nit: "used first" I don't think is semantically correct. I think a better phrase would be "takes precedence over a users metadata".

Reply:
Changed